src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -1,182 +1,294 @@
     }
+    if (from === FROM_FOLLOWED_DOMAIN) {
+      console.log('from 1');
+      const newFollowedDomains = [...followedDomains];
+      newFollowedDomains[index].following = !!willFollow;
+      newFollowedDomains[index].user_id_follower = myId;
 
-    const __renderDiscoveryItem = (from, key, item, index) => <View key={`${key}-${index}`} style={styles.domainContainer}>
-        <DomainList isDomain={true}
+      setFollowedDomains(newFollowedDomains);
+    }
+
+    if (from === FROM_UNFOLLOWED_DOMAIN) {
+      console.log('from 2');
+      const newUnfollowedDomains = [...unfollowedDomains];
+      newUnfollowedDomains[index].following = !!willFollow;
+      newUnfollowedDomains[index].user_id_follower = myId;
+
+      setUnfollowedDomains(newUnfollowedDomains);
+    }
+    const data = {
+      domainId: item.domain_id_followed,
+      source: 'discoveryScreen'
+    };
+    if (willFollow) {
+      await followDomain(data);
+    } else {
+      await unfollowDomain(data);
+    }
+    if (searchText.length > 0) fetchData();
+  };
+
+  const __renderDiscoveryItem = (from, key, item, index) => {
+    return (
+      <View key={`${key}-${index}`} style={styles.domainContainer}>
+        {route.name === 'Followings' && item.user_id_follower !== null && (
+          <DomainList
+            isDomain={true}
             onPressBody={() => __handleOnPressDomain(item)}
             handleSetFollow={() => __handleFollow(from, true, item, index)}
             handleSetUnFollow={() => __handleFollow(from, false, item, index)}
             item={{
-                name: item.domain_name,
-                image: item.logo,
-                isunfollowed: item.user_id_follower === null,
-                description: item.short_description || null
+              name: item.domain_name,
+              image: item.logo,
+              isunfollowed: !item.following,
+              description: item.short_description || null
             }}
-        />
-    </View>
-
-    const __renderDomainItems = () => {
-        if (isFirstTimeOpen) return [<DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />].concat(domains.map((item, index) => __renderDiscoveryItem(FROM_FOLLOWED_DOMAIN_INITIAL, "followedDomainDiscovery", { ...item, user_id_follower: item.user_id_follower }, index)))
+          />
+        )}
+        {route.name !== 'Followings' && (
+          <DomainList
+            isDomain={true}
+            onPressBody={() => __handleOnPressDomain(item)}
+            handleSetFollow={() => __handleFollow(from, true, item, index)}
+            handleSetUnFollow={() => __handleFollow(from, false, item, index)}
+            item={{
+              name: item.domain_name,
+              image: item.logo,
+              isunfollowed: !item.following,
+              description: item.short_description || null
+            }}
+          />
+        )}
+      </View>
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is duplicated code for rendering the `DomainList` component. This could be simplified by removing the conditional rendering since the props passed to the component are the same in both cases:

```diff
- {route.name === 'Followings' && item.user_id_follower !== null && (
-   <DomainList
-     isDomain={true}
-     onPressBody={() => __handleOnPressDomain(item)}
-     handleSetFollow={() => __handleFollow(from, true, item, index)}
-     handleSetUnFollow={() => __handleFollow(from, false, item, index)}
-     item={{
-       name: item.domain_name,
-       image: item.logo,
-       isunfollowed: !item.following,
-       description: item.short_description || null
-      }}
-   />
- )}
- {route.name !== 'Followings' && (
-   <DomainList
-     isDomain={true}
-     onPressBody={() => __handleOnPressDomain(item)}
-     handleSetFollow={() => __handleFollow(from, true, item, index)}
-     handleSetUnFollow={() => __handleFollow(from, false, item, index)}
-     item={{
-       name: item.domain_name,
-       image: item.logo,
-       isunfollowed: !item.following,
-       description: item.short_description || null
-     }}
-   />
- )}

+ <DomainList
+   isDomain={true}
+   onPressBody={() => __handleOnPressDomain(item)}
+   handleSetFollow={() => __handleFollow(from, true, item, index)}
+   handleSetUnFollow={() => __handleFollow(from, false, item, index)}
+   item={{
+     name: item.domain_name,
+     image: item.logo,
+     isunfollowed: !item.following,
+     description: item.short_description || null
+    }}
+ />
```

This will make the code cleaner and easier to maintain.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
in following and discovery has different behavior to show the item, so it need to distinguish one from another